apps/sms/js/navigation.js

```diff
@@ -116,6 +116,17 @@ var Navigation = {
   },
 
   /**
+   * Ensures that current panel is correctly set, if no current panel set and
+   * we're not in the process of transitioning then set default panel as current
+   * one.
+   */
+  ensureCurrentPanel: function() {
```

Comment:
I'm not sure about the correct name for this method, but the idea is to fix app navigation state when it's not valid and there is a chance that app instance will be used by user (I'll add unit tests once we figure out best approach to deal with the issue). As you found on Friday, my solution doesn't work correctly when notification click is handled by system message + it wouldn't work correctly when you have already opened app (navigated to Composer for example) and then you remove notification (that wasn't created by current app instance, so click will be propagated through system message) - user will be navigated to thread list :)

Initially I put this code into ActivityHandler, but it uses only Navigation internal data, so I moved it to Navigation.js.

So what are your thoughts about that?